### PR TITLE
[MI-3282] Added logic to expand issue in comment webhook notification

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
Link: https://github.com/mattermost/mattermost-plugin-jira/pull/953#issuecomment-1604073508

### CODE REVIEW

File: server/issue.go [hunk 1]
```diff
@@ -736,6 +736,32 @@ func getIssueCustomFieldValue(issue *jira.Issue, key string) StringSet {
 	return nil
 }
 
+func (p *Plugin) getIssueDataForCloudWebhook(instance Instance, issueKey string) (*jira.Issue, error) {
+	ci, ok := instance.(*cloudInstance)
+	if !ok {
+		return nil, errors.Errorf("Must be a JIRA Cloud instance, is %s", instance.Common().Type)
```
Comment (Kshitij-Katiyar) [inline server/issue.go:742]: error should start with lower case

File: server/issue.go [hunk 2]
```diff
@@ -736,6 +736,32 @@ func getIssueCustomFieldValue(issue *jira.Issue, key string) StringSet {
...
+	}
+
+	issue, resp, err := jiraClient.Issue.Get(issueKey, nil)
+	if err != nil {
+		switch {
+		case resp == nil:
+			return nil, errors.WithMessage(userFriendlyJiraError(nil, err),
+				"request to Jira failed")
+
+		case resp.StatusCode == http.StatusNotFound || resp.StatusCode == http.StatusUnauthorized:
```
Comment (Kshitij-Katiyar) [inline server/issue.go:757]: rather combining these make 2 different cases

File: server/webhook_jira.go [hunk 3]
```diff
@@ -29,6 +31,50 @@ type JiraWebhook struct {
...
+	if isCommentEvent && instance.Common().IsCloudInstance() {
+		if _, ok := instance.(*cloudInstance); ok {
+			issue, err := p.getIssueDataForCloudWebhook(instance, jwh.Issue.ID)
+			if err != nil {
+				return err
+			}
+
+			jwh.Issue = *issue
+		} else if _, ok := instance.(*cloudOAuthInstance); ok {
+			mmID, err := p.userStore.LoadMattermostUserID(instanceID, jwh.Comment.Author.AccountID)
```
Comment (Kshitij-Katiyar) [inline server/webhook_jira.go:51]:
```suggestion
			mmUserID, err := p.userStore.LoadMattermostUserID(instanceID, jwh.Comment.Author.AccountID)
```

File: server/webhook_jira.go [hunk 4]
```diff
@@ -29,6 +31,50 @@ type JiraWebhook struct {
...
+			if err != nil {
+				return err
+			}
+
+			c, err := p.userStore.LoadConnection(instance.GetID(), mmID)
```
Comment (Kshitij-Katiyar) [inline server/webhook_jira.go:56]:
```suggestion
			conn, err := p.userStore.LoadConnection(instance.GetID(), mmID)
```

File: server/issue.go (hunk 2 above)
Comment (raghavaggarwal2308) [reply server/issue.go:757]: This is an existing code. We removed it a while back, now we added it again.

File: server/issue.go [hunk 5]
```diff
@@ -736,6 +736,32 @@ func getIssueCustomFieldValue(issue *jira.Issue, key string) StringSet {
...
+	jiraClient, err := ci.getClientForBot()
+	if err != nil {
+		return nil, err
+	}
+
+	issue, resp, err := jiraClient.Issue.Get(issueKey, nil)
+	if err != nil {
+		switch {
+		case resp == nil:
+			return nil, errors.WithMessage(userFriendlyJiraError(nil, err),
```
Comment (Kshitij-Katiyar) [inline server/issue.go:754]: same line

**Review by Kshitij-Katiyar — APPROVED** [review]

**Pull request merged**